src/sql/workbench/contrib/notebook/browser/notebook.contribution.ts

```diff
@@ -182,7 +182,7 @@ CommandsRegistry.registerCommand({
 		for (let editor of editors) {
 			if (editor instanceof NotebookInput) {
 				let model: INotebookModel = editor.notebookModel;
-				if (model.providerId === 'jupyter') {
+				if (model.providerId === 'jupyter' && model.clientSession.isReady) {
```

Comment:
So this will restart it if the session is ready - but what about sessions that are still in the process of starting? Wouldn't that possibly then be using the wrong python installation?

Reply:
Yes, but I think it would be rare for a user to open a Python notebook, open up the configure python dialog and finish installing a new Python version before the original notebook session is done loading.

Reply:
@chlafreniere FYI. I'll leave it to you two to decide whether this is something that needs to be handled